openslides/core/static/js/core/base.js

```diff
@@ -860,6 +860,74 @@ angular.module('OpenSlidesApp.core', [
     }
 ])
 
+.factory('Fonts', [
+    'Config',
+    'gettext',
+    function (Config, gettext) {
+        var extensionFormatMap = {
+            'ttf': 'truetype',
+            'woff': 'woff',
+        };
+
+        return {
+            getKeys: function () {
+                return Config.get('fonts_available').value;
+            },
+            getAll: function () {
+                var self = this;
+                return _.map(this.getKeys(), function (key) {
+                    return self.get(key);
+                });
+            },
+            get: function (key) {
+                var config = Config.get(key);
+                if (config) {
+                    config.value.key = key;
+                    return config.value;
+                }
+            },
+            getUrl: function (key) {
+                var font = this.get(key);
+                if (font) {
+                    var path = font.path;
+                    if (!path) {
+                        return font.default;
+                    }
+                    return path;
+                }
+            },
+            getForCss: function (key) {
+                var url = this.getUrl(key);
+                if (url) {
+                    var ext = _.last(url.split('.'));
+                    return "url('" + url + "') format('" +
+                        extensionFormatMap[ext] + "')";
+                }
+            },
+            isMediafileUsedAsFont: function (mediafile) {
```

Comment:
Hm. Is it possible not to have mediafile code in core/base?

Reply:
Done. Refactored logo and font code.